etcdserver/server.go

```diff
@@ -632,6 +628,21 @@ func (s *EtcdServer) Lead() uint64 { return atomic.LoadUint64(&s.r.lead) }
 
 func (s *EtcdServer) Leader() types.ID { return types.ID(s.Lead()) }
 
+// RequestTimeout returns an appropriate timeout to wait for request to be committed.
+func (s *EtcdServer) RequestTimeout() time.Duration {
```

Comment:
I think this over complicated the issue. We do not want a exact value. We just need a max timeout.